src/pages/api/howManyProducts.ts

```diff
@@ -0,0 +1,44 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { NUMBER_OF_PRODUCTS_COOKIE } from '../../constants/index';
+import { getDomain, setCookieOnResponseObject, redirectToError, redirectTo } from './apiUtils';
+import { isSessionValid } from './service/validator';
+import { InputCheck } from '../howManyProducts';
+
+export const isNumberOfProductsInvalid = (req: NextApiRequest): InputCheck => {
+    const { numberOfProductsInput = '' } = req.body;
+    const inputAsNumber = Number(numberOfProductsInput);
+    let error;
+    if (numberOfProductsInput === '' || Number.isNaN(inputAsNumber)) {
+        error = 'Enter a number';
+    } else if (!Number.isInteger(inputAsNumber) || Number(inputAsNumber) > 10 || Number(inputAsNumber) < 1) {
```

Comment:
can we not use the inputAsNumber variable here as we are doing the number conversion again?

Reply:
Can do! I've updated this for the 'greater than' and 'less than' comparisons but the isInteger check is a method on the Number constructor so looks like we'll need to keep this in